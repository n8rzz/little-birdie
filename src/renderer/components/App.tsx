import React, { useEffect, useCallback } from 'react';
import { getFeedList } from '../stores/feed/feed.service'
import { IFeed } from '../stores/feed/models/i-feed'
import { FeedList } from './feed/FeedList';
import { Header } from './Header';
import { Footer } from './Footer';

interface IProps {}

export const App: React.FC<IProps> = () => {
  // const [items, setItems] = React.useState<IFeed[]>([]);

  useEffect(() => getParsedFeedRss);

  const getParsedFeedRss = useCallback(async () => {
    console.log('===');
    Promise.resolve(true);
    // const feeds = await getFeedList();

    // setItems(feeds);
  }, []);

  console.log('$$$');

  return (
    <div className="container">
      <Header />
      <FeedList items={['Github', 'travis', 'heroku']} />
      <Footer />
    </div>
  );
}
