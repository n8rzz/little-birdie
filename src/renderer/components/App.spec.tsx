import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';

import { App } from './App';

describe('App', () => {
  it('should match snapshot', () => {
    const tree = TestRenderer.create(<App />);

    expect(tree).toMatchSnapshot();
  });
});
