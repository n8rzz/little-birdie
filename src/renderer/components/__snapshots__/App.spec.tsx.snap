// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="container"
>
  <header>
    Little Birdy
  </header>
  <section>
    <ul>
      <li
        className="providerList-item"
      >
        <a
          href="#"
        >
          heroku
        </a>
      </li>
      <li
        className="providerList-item"
      >
        <a
          href="#"
        >
          heroku
        </a>
      </li>
      <li
        className="providerList-item"
      >
        <a
          href="#"
        >
          heroku
        </a>
      </li>
    </ul>
  </section>
  <footer>
    Sync
  </footer>
</div>
`;
